File: scrapers/google_maps/models/review.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, HttpUrl

class Review(BaseModel):
    review_source: str = "GOOGLE"
    target_id: int
    target_type: str
    content: Optional[str] = None
    overall_rating: int
    google_review_id: str
    google_author_name: str
    google_author_url: HttpUrl
    google_author_photo_url: HttpUrl
    published_time: datetime

File: frontend/src/types/review.ts
export enum ReviewSource {
  GOOGLE = "GOOGLE",
  IN_APP = "IN_APP",
}

export enum ReviewTargetType {
  DEMENTIA_DAY_CARE = "CARESERVICE::DEMENTIA_DAY_CARE",
  DEMENTIA_HOME_CARE = "CARESERVICE::DEMENTIA_HOME_CARE",
}

export type Review = {
  reviewSource: ReviewSource;
  targetId: number;
  targetType: ReviewTargetType;
  overallRating: number;
  content: string | null;
  authorId: string | null;
  authorName: string;
  googleReviewId: string;
  googleAuthorUrl: string;
  googleAuthorPhotoUrl: string;
  publishedTime: string;
};

export type ReviewCreate = {
  review_source: ReviewSource;
  target_id: number;
  target_type: ReviewTargetType;
  overall_rating: number;
  author_name: string;
  content: string;
};
